mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -163,12 +163,20 @@ class RFormula @Since("1.5.0") (@Since("1.5.0") override val uid: String)
     }.toMap
 
     // Then we handle one-hot encoding and interactions between terms.
+    var hasReferenceCategory = false
```

Comment:
I rename ```hasReferenceCategory``` to ```keepReferenceCategory``` with default initial value ```false```. This is because if users fit with intercept, it will not trigger ```keepReferenceCategory```, so ```keepReferenceCategory``` always as ```false```. @actuaryzhang I think ```firstString``` is not an expressive name, if users fit with intercept, ```RFormula``` will check the first string as well, but doesn't keep all categories. So after run pass this code snippet, ```firstString``` is still ```true```